orcestra_book/plans/HALO-20240903a.md

```diff
@@ -0,0 +1,318 @@
+    if (index[0]+index[-4:]!='c_out'):
+        print(f"{index:12s} {LatLon(row['lat'],row['lon']).format_pilot():20s}, FL{int(row['fl']):03d}, {takeoff_time+row['duration']:%H:%M:%S}, {notes.get(index,'')}" )
+print ('\n-- circle centers:')
+for point in waypoints:
+    if isinstance(point, IntoCircle):
+        point = point.center
+        print (f'{point.label:12s} {point.format_pilot()}')
+print ('\n-- extra waypoints:')
+for point in extra_waypoints:
+    print (f'{point.label:12s} {point.format_pilot()}, {notes.get(point.label,'')}' )
+print ('\nCrew:')
+for position,person in crew.items():
+    print (f'{position:22s} {person}')
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+vertical_preview(waypoints)
+plt.title("Profile")
+```
+
+```{code-cell} ipython3
+import os
+
+localdir = "/Users/m219063/Downloads/"
+if os.path.isdir(localdir):
+    print (localdir)
+    with open(f"/Users/m219063/Downloads/{flight_index}.kml", "w") as f:
+        f.write(to_kml(path))
+        
+    with open( f"/Users/m219063/Downloads/{flight_index}_waypoints.txt", "w") as file:
+        file.write(f"Flight {flight_index}\n\n")
+        #
+        # DM Format
+        file.write("------------------------------------------------------------\n")
+        file.write("\nDM Format:\n")
+        file.write(" ".join(wp.format_1min() for wp in waypoint_centers) + "\n")
+        for point in extra_waypoints:
+            file.write(f"Extra waypoint: {point.format_1min()}\n")
+        #
+        # DM.mm format
+        file.write("\n------------------------------------------------------------\n")
+        file.write("\nDMmm Format:\n")
+        for point in waypoint_centers:
+            file.write(f"{point.format_pilot()}, {point.label}\n")
+        file.write("\n-- extra waypoints:\n")
+        for point in extra_waypoints:
+            file.write(f"{point.format_pilot()}, {notes.get(point.label,'')}\n")
+        #
+        # Detailed overview with notes
+        file.write("\n------------------------------------------------------------\n")
+        file.write(f"\n\nDetailed Overview:\n")
+        for index, row in plan.iterrows():
+            if (index[0]+index[-4:]!='c_out'):
+                file.write(f"{index:12s} {LatLon(row['lat'],row['lon']).format_pilot():20s}, FL{int(row['fl']):03d}, {takeoff_time+row['duration']:%H:%M:%S}, {notes.get(index,'')}\n" )
+        file.write ('\n -- circle centers:')
+        for point in waypoints:
+            if isinstance(point, IntoCircle):
+                point = point.center
+                file.write (f'\n{point.label:12s} {point.format_pilot()}')
+        file.write ('\n\n -- extra waypoints:')
+        for point in extra_waypoints:
+            file.write (f'\n{point.label:12s} {point.format_pilot()}, {notes.get(point.label,'')}' )
+        file.write ('\n\nCrew:')
+        for position,person in crew.items():
+            file.write (f'\n{position:22s} {person}')
+```
+
+```{code-cell} ipython3
+
```

Comment:
You might want to check the other flight plans on our website, which include kml and geojson export, and add that functionality here too.

Reply:
It's about this line: https://github.com/orcestra-campaign/book/blob/27e4604e0a9da8ecbcede7964825b9dbf1a8e797/orcestra_book/plans/HALO-20240825a.md?plain=1#L237

but now apparently, the PR exploded...

Reply:
okay, in the meantime it seems new things appeared there, which are now in.. btw for her plan one should use flight_index, rather than rewrite the date.